src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -881,5 +881,30 @@ public static CurrentProcessingFileInformation convertFromMapOfObjects(
                             .get(PERSISTED_CURRENT_PROCESSING_FILE_HASH)))
                     .build();
         }
+
+        private String getFileHashFromTopic(Topic topic) {
+            Topics topics = topic.parent;
+            Topic hashTopic = topics.find(PERSISTED_CURRENT_PROCESSING_FILE_HASH);
+
+            if (hashTopic != null) {
+                return Coerce.toString(hashTopic);
```

Comment:
this may return `null` is that OK and handled correctly by consumers?

Reply:
https://github.com/aws-greengrass/aws-greengrass-log-manager/blob/main/src/main/java/com/aws/greengrass/logmanager/LogManagerService.java#L415
It is handled here.